### PR TITLE
HOSTEDCP-1813: Moves MachineIdentity from the HostedCluster API to the NodePool API

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
Moves `MachineIdentity` from the Azure section in the HostedCluster API to the Azure section in the NodePool API

**Which issue(s) this PR fixes**:
Fixes [HOSTEDCP-1813](https://issues.redhat.com/browse/HOSTEDCP-1813)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [x] This change includes unit tests.

### CODE REVIEW

File: api/hypershift/v1beta1/nodepool_types.go [hunk 1]
```diff
@@ -964,6 +964,11 @@ type AzureNodePoolPlatform struct {
 	// If not specified then Boot diagnostics will be disabled.
 	// +optional
 	Diagnostics *Diagnostics `json:"diagnostics,omitempty"`
+
+	// MachineIdentityID is used as the user-assigned identity to be assigned to the VMs
+	//
+	// +optional
+	MachineIdentityID string `json:"machineIdentityID,omitempty"`
```
Comment (miguelsorianod) [inline api/hypershift/v1beta1/nodepool_types.go:971]: Could we document what is the behavior if `MachineIdentityID` is not provided?

Could we document if the resource associated to the ID needs to preexist beforehand in Azure?

Thanks

File: api/hypershift/v1alpha1/hostedcluster_types.go [hunk 2]
```diff
@@ -1573,7 +1573,6 @@ type AzurePlatformSpec struct {
 	VnetID            string `json:"vnetID"`
 	SubnetID          string `json:"subnetID"`
 	SubscriptionID    string `json:"subscriptionID"`
-	MachineIdentityID string `json:"machineIdentityID"`
```
Comment (miguelsorianod) [inline api/hypershift/v1alpha1/hostedcluster_types.go:1576]: So, machine identity is not needed at all during HostedCluster creation?

File: api/hypershift/v1beta1/nodepool_types.go (hunk 1 above)
Comment (bryan-cox) [reply api/hypershift/v1beta1/nodepool_types.go:971]: Yeah I think we have discussed what happens if it is not provided a few times. 

I will add it explicitly here though, but this field could change as workload identity/MSI work progresses.

File: api/hypershift/v1alpha1/hostedcluster_types.go (hunk 2 above)
Comment (bryan-cox) [reply api/hypershift/v1alpha1/hostedcluster_types.go:1576]: No

File: api/hypershift/v1beta1/nodepool_types.go (hunk 1 above)
Comment (miguelsorianod) [reply api/hypershift/v1beta1/nodepool_types.go:971]: > Yeah I think we have discussed what happens if it is not provided a few times.

Yes, I am aware :+1: . It is more for consumers of the API and other people that might not be aware of it.

**Pull request merged**